pkg/watchdog/pkg_test.go

```diff
@@ -4,8 +4,374 @@
 
 package watchdog
 
-import "testing"
+import (
+	"fmt"
+	"io/ioutil"
+	"testing"
+	"time"
 
-func TestTODO(t *testing.T) {
-	// TODO: Write a unit test.
+	"golang.org/x/sys/unix"
+)
+
+func getTestStruct() (*Watchdog, error) {
+	f, err := ioutil.TempFile("", "testfile-")
+	if err != nil {
+		return nil, err
+	}
+	return &Watchdog{
+		File:               f,
+		unixSyscall:        nil,
+		unixIoctlGetUint32: nil,
+	}, nil
+}
+
+func TestOpen(t *testing.T) {
+	f, err := ioutil.TempFile("", "testfile-")
+	if err != nil {
+		t.Error(err)
+	}
+	defer f.Close()
+
+	w, err := Open(f.Name())
+	if err != nil {
+		t.Error(err)
+	}
+	defer w.Close()
+}
+
+func TestOpenError(t *testing.T) {
+	_, err := Open("testfile")
+	if err == nil {
+		t.Error("Open succeeded but should have failed")
+	}
+}
+
+func TestSupport(t *testing.T) {
```

Comment:
This is very impressive but it's a gigantic amount of code, and something about it makes me feel it can be much more simply done as a table driven test

func TestThing(t*testing.T) {
okSysCall = func(_ trap, _ ...int) (uintptr, uintpre, unix.Errno) {
return 0, 0 0
)
define other system calls as needed
then
for tt = range []struct {
call func(trap, ...int) (uintptr, uintptr, unix.Errno)
operation
}{
{okSysCall, op}
...
}{
w, err := getTestStruct()
if err != nil {
t.Errorf(...)
}
tt.okSysCall()

and so on

I am almost certain this can be set up as a table with a bit of thought, and we can greatly reduce the size of this function. 

Again, let's iterate on this one PR and get it to a good place, and that will let us more easily get the others done -- we'll have established some patterns.

Thanks.

Reply:
Take a look at how it's done here: https://github.com/u-root/u-root/blob/master/pkg/spidev/spidev_linux_test.go#L105